File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.cpp
#define ll long long
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        ll sum=0;
        ll n=nums.size();
        vector<ll>ps(n);
        for(ll i=0;i<n;i++){
            sum+=nums[i];
            ps[i]=(sum);
        }
        for(ll i=n-1;i>=2;i--){
            if(ps[i-1]>nums[i])return ps[i-1]+nums[i];
        }
        return -1;
    }
};

File: 2971-find-polygon-with-the-largest-perimeter/2971-find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int n=nums.length;
        long[] ps=new long[n];
        long cs=0;
        for(int i=0;i<n;i++){
            cs+=nums[i];
            ps[i]=cs;
        }
        for(int i=n-1;i>=2;i--){
            if(nums[i]<ps[i-1])return nums[i]+ps[i-1];
        }
        return -1;
    }
}